src/ess/powder/conversion.py

```diff
@@ -211,11 +218,44 @@ def convert_to_dspacing(
         for key in ('wavelength', 'two_theta'):
             if key in out.coords.keys():
                 out.coords.set_aligned(key, False)
-    out.bins.coords.pop('tof', None)
+    # out.bins.coords.pop('tof', None)
     out.bins.coords.pop('wavelength', None)
     return DspacingData[RunType](out)
 
 
+def assemble_output_calibration(
+    data: DspacingData[SampleRun],
+) -> OutputCalibrationData:
+    # Use nanmean because pixels without events have position=NaN.
+    average_l = sc.nanmean(data.coords["Ltotal"])
+    average_two_theta = sc.nanmean(data.coords["two_theta"])
+    difc = sc.to_unit(
+        2
+        * sc.constants.m_n
+        / sc.constants.h
+        * average_l
+        * sc.sin(0.5 * average_two_theta),
+        unit='us / angstrom',
+    ).value
+    return OutputCalibrationData(
+        sc.DataArray(
+            sc.array(dims=['calibration'], values=[difc]),
```

Comment:
Why are why taking the value (stripping units?) and put it back into a variable?

Reply:
Because the underlying CIF writer wants the coefficients as a data array (https://scipp.github.io/scippneutron/generated/modules/scippneutron.io.cif.CIF.html#scippneutron.io.cif.CIF.with_powder_calibration):
```python
cal = sc.DataArray(
    sc.array(dims=['cal'], values=[3.4, 0.2]),
    coords={'power': sc.array(dims=['cal'], values=[0, 1])},
)
```
It is unfortunate that we have to remove the unit but that is how CIF wants the data.

We could instead keep a dict until the last moment where we need to convert to the above data array:
```python
{
   0: sc.scalar(3.4, unit='us'),
   1: sc.scalar(0.2, unit='us/angstrom')
}
```
Would you prefer this?

Reply:
I don't mind much, but can you explain what is going on in the comments?

Reply:
Have a look at the new implementation.